components/reusable-components/navigation/Navigation.tsx

```diff
@@ -1,36 +1,72 @@
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+
 'use client';
 
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+
+// eslint-disable-next-line import/no-duplicates
 import Link from 'next/link';
+import Image from 'next/image';
 import styles from './navigation.module.scss';
 import Button from '../button/Button';
+import { useTheme } from '../../context/ThemeContext';
+// eslint-disable-next-line import/no-unresolved
+import solImage from '../../../public/sol.png';
+// eslint-disable-next-line import/no-unresolved
+import moonImage from '../../../public/lune.png';
 
 const Navigation = () => {
+  const { theme, toggleTheme } = useTheme();
+  const [isSun, setIsSun] = useState(true);
+
+  useEffect(() => {
+    document.body.className = theme;
+  }, [theme]);
+  useEffect(() => {
+    const storedIsSun = localStorage.getItem('isSun');
+    if (storedIsSun !== null) {
+      setIsSun(storedIsSun === 'true');
+    }
+  }, []);
+  const handleClick = () => {
+    toggleTheme();
+    setIsSun((prevIsSun) => {
+      const newIsSun = !prevIsSun;
+      localStorage.setItem('isSun', String(newIsSun));
+      return newIsSun;
+    });
+  };
+
```

Comment:
Since `isSun` is true for light theme and false for dark theme, we can eliminate the isSun state and we can derive `isSun` value directly from the theme.
This reduces the component's state complexity and keeps your UI in sync with the theme state.

- **handleClick function** should just toggle the theme. We should let the theme context handle everything, including localStorage.

- and we might need an useEffect here that would apply theme class to body (we maybe should move that to higher level)